### ISSUE TITLE
[Logging V2] Windows node taints and tolerations.

### ISSUE THREAD

**Issue by paynejacob** [rancher/rancher#28720]
Ensure the logging operator and any containers it creates are not schedule on windows nodes.

**Comment by nickgerace** [comment]
Using AMI: Microsoft Windows Server 2019 Base with Containers

**Comment by nickgerace** [comment]
We potentially need...

1) Tolerance on the Linux node
2) Node selector that only selects Linux nodes

**Comment by nickgerace** [comment]
The PR is complete for adding tolerations and node selector settings to all "logging" pods. However, one upstream bug has been found...

**Bug Description**: L204 of the below file shows that the fluentd-configcheck container does not inherit the tolerances and nodeselector settings from the logging-operator.
**Impact**: Fortunately, the PR to add tolerations and nodeselector settings is still valid on Rancher's side. However, the core issue of running pods on Windows nodes will not be fixed so long as the upstream bug exists.
**Relevant Link**: [https://github.com/banzaicloud/logging-operator/blob/73030d200f6f02c6ce9c174fb34eb149db1312e0/pkg/resources/fluentd/appconfigmap.go#L204](https://github.com/banzaicloud/logging-operator/blob/73030d200f6f02c6ce9c174fb34eb149db1312e0/pkg/resources/fluentd/appconfigmap.go#L204)

**Comment by nickgerace** [comment]
Created an issue with the banzaicloud/logging-operator repo: [https://github.com/banzaicloud/logging-operator/issues/592](https://github.com/banzaicloud/logging-operator/issues/592)

**Comment by nickgerace** [comment]
There is a PR out for the upstream fix: [https://github.com/banzaicloud/logging-operator/pull/594](https://github.com/banzaicloud/logging-operator/pull/594)

**Comment by izaac** [comment]
Rancher version `master-head (10/15/2020)` _6c00180_

Logging operator fails to pull image as it lands in a Windows Node.

**Comment by cbron** [comment]
Can we re-test this, and if it fails again get the creds to the box. I think it was likely a syncing error. The chart version should be 3.6.001

**Comment by izaac** [comment]
@cbron I see the chart version `v3.6.001` and it is now deploying the logging operator to the Linux worker

**Comment by izaac** [comment]
`master-head (10/16/2020)` _cd2258c_
I see the logging operator landing in the Linux worker however all the confichecks in `ImagePullBackOff` status `rancher/banzaicloud-fluentd:v1.11.2-alpine-2`. 

No fluentbit workloads are created.

**Comment by nickgerace** [comment]
That is the upstream bug I noted above: https://github.com/rancher/rancher/issues/28720#issuecomment-708687880

All those of those pods are configcheck pods, and were previously unable to inherit their logging-operator pod's settings. Fortunately, the fix was merged ~7 hours ago: https://github.com/banzaicloud/logging-operator/pull/594

I'd be happy to open a PR to update the image tag, but the latest release is still 3.6.0. We may either have to ask for a custom release, pull our own customer image, or wait until the newest release. (@cbron: do you have a preference here?)

**Comment by cbron** [comment]
We'll wait for a new banzai release. In the meantime I suppose we have to bump this to 2.5.3 until that comes out.

**Comment by nickgerace** [comment]
The new 3.7.0 chart successfully fixed the issue.
```
kubectl get pods -A
NAMESPACE               NAME                                           READY   STATUS      RESTARTS   AGE
cattle-logging-system   rancher-logging-dff45796d-7jhtr                1/1     Running     0          30s
cattle-logging-system   rancher-logging-fluentbit-82pbw                1/1     Running     0          13s
cattle-logging-system   rancher-logging-fluentbit-tft6c                1/1     Running     0          13s
cattle-logging-system   rancher-logging-fluentd-0                      2/2     Running     0          13s
cattle-logging-system   rancher-logging-fluentd-configcheck-c3054702   0/1     Completed   0          25s
```

Pending merge of PR to move to QA: https://github.com/rancher/charts/pull/817

**Comment by nickgerace** [comment]
Merged in 3.7.3: https://github.com/rancher/charts/pull/817

Essentially, these series of changes ensure that logging related pods do not run on Windows nodes, and only run on Linux nodes.

**Comment by izaac** [comment]
Rancher version `v2.5-head (11/17/2020)` _9a67809_
Rancher version `master-head (11/17/2020)` _90cdb6d6d_

Logging chart version: `3.7.301`

I was able to execute P1 test cases on a windows cluster. Fluentbit is no longer trying to pull images or create pods in the Windows nodes of a Custom RKE cluster.
